app/registries/foreman/settings/auth.rb

```diff
@@ -0,0 +1,160 @@
+Foreman::SettingManager.define(:foreman) do
+  category(:auth, N_('Authentication')) do
+    setting('oauth_active',
+      type: :boolean,
+      description: N_("Foreman will use OAuth for API authorization"),
+      default: false,
+      full_name: N_('OAuth active'))
+    setting('oauth_consumer_key',
+      type: :string,
+      description: N_("OAuth consumer key"),
+      default: '',
+      full_name: N_('OAuth consumer key'),
+      encrypted: true)
+    setting('oauth_consumer_secret',
+      type: :string,
+      description: N_("OAuth consumer secret"),
+      default: '',
+      full_name: N_("OAuth consumer secret"),
+      encrypted: true)
+    setting('oauth_map_users',
+      type: :boolean,
+      description: N_("When enabled, Foreman will map users by username in request-header. If this is disabled, OAuth requests will have admin rights."),
+      default: true,
+      full_name: N_('OAuth map users'))
+    setting('failed_login_attempts_limit',
+      type: :integer,
+      description: N_("Foreman will block user login after this number of failed login attempts for 5 minutes from offending IP address. Set to 0 to disable bruteforce protection"),
```

Comment:
Perhaps out of scope for this PR, but I think this description has a better flow.
```suggestion
      description: N_("Foreman will block user logins from an IP address after this number of failed login attempts for 5 minutes. Set to 0 to disable bruteforce protection"),
```